### PR TITLE
Adding sync trigger endpoint for Kubernetes managed K8SE environment.

### PR DESCRIPTION
**Pull request by SatishRanjan**
This PR enables the functionality for the function host to call the build service endpoint running in the managed kubernetes K8SE environment to sync the triggers for KEDA scaling to work as expected. 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs [hunk 1]
```diff
@@ -558,6 +563,10 @@ internal HttpRequestMessage BuildSetTriggersRequest()
 
             var hostname = _hostNameProvider.Value;
             var url = $"{protocol}://{hostname}/operations/settriggers";
+            if (_environment.IsKubernetesManagedHosting())
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs:558]: nit: can you make this an if/else, so we don't form the url above only to overwrite it?

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs [hunk 2]
```diff
@@ -558,6 +563,10 @@ internal HttpRequestMessage BuildSetTriggersRequest()
...
+            {
+                url = $"http://{ManagedKubernetesBuildServiceName}.{ManagedKubernetesBuildServiceNamespace}.svc.cluster.local:{ManagedKubernetesBuildServicePort}/operations/settriggers";
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs:560]: What is the auth model for this request? In SetTriggers below, we're sending a token that the FE uses for authentication. Is this request also handled by the FE?

Comment (SatishRanjan) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs:560]: Hi @mathewc - in this case request goes to the build service (kudulite) running in the same Kubernetes cluster as the function host, and the encryption key for the function app will also be accessible to build service, which is used to decrypt the token and pass the authentication successfully. The function app access token is created/updated through the geo-master control plane code as a Kubernetes secret and build service will have access to read the secret. So, this flow in Kubernetes doesn't hit the Antares front end. Hope this helps.

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs:560]: Ok, so the token is actually required (not ignored) in your scenario - that's all I was trying to establish.

Comment (SatishRanjan) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs:560]: Got it, yes that's right @mathewc.

**Review by mathewc — DISMISSED** [review]

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs [hunk 3]
```diff
@@ -291,8 +291,17 @@ public static bool IsAppService(this IEnvironment environment)
...
         {
-            return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(KubernetesServiceHost))
+            try
+            {
+                return !string.IsNullOrEmpty(environment.GetEnvironmentVariable(KubernetesServiceHost))
                 && !string.IsNullOrEmpty(environment.GetEnvironmentVariable(PodNamespace));
+            }
+            catch
+            {
+                //if there's an error reading the environment variable then return false
```
Comment (mathewc) [inline src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: I'd rather you just fix the test to mock things appropriately. Let's not have test code here.

**Comment by SatishRanjan** [comment]
@mathewc - could you please take a look again? I've resolved your comments.

**Review by mathewc — DISMISSED** [review]

**Comment by pragnagopa** [comment]
@SatishRanjan  - JIT Trace is updated. Will merge the PR once the build is green.

**Review by pragnagopa — APPROVED** [review]

**Comment by SatishRanjan** [comment]
Thanks a ton @pragnagopa :)

**Pull request merged**